.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+# Enable with the following commands:
+#   python3 -m pip install pre-commit
+#   pre-commit install
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```

Comment:
@bossie Pre commit ok? It should only do some light formatting on added files (removing trailing whitespace and makes files end with a newline). 
As a test I let ran on all files and `mvn clean install -Dmaven.test.skip=true` still worked